fmlloader/src/main/java/net/minecraftforge/fml/loading/FMLConfig.java

```diff
@@ -15,21 +17,77 @@
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.function.Function;
 
 import static net.minecraftforge.fml.loading.LogMarkers.CORE;
 
 public class FMLConfig
 {
+    public enum ConfigValue {
+        SPLASHSCREEN("splashscreen", Boolean.TRUE, "should we show the early splashscreen"),
+        MAX_THREADS("maxThreads", -1, "Max threads for early initialization parallelism,  -1 is based on processor count", FMLConfig::maxThreads),
+        VERSION_CHECK("versionCheck", Boolean.TRUE, "Enable forge global version checking"),
+        DEFAULT_CONFIG_PATH("defaultConfigPath", "defaultconfigs", "Default config path for servers"),
+        DISABLE_OPTIMIZED_DFU("disableOptimizedDFU", Boolean.TRUE, "Disables Optimized DFU client-side - already disabled on servers"),
+        EARLY_WINDOW_PROVIDER("earlyWindowProvider", "fmlearlywindow", "Early window provider"),
+        EARLY_WINDOW_WIDTH("earlyWindowWidth", 854, "Early window width"),
+        EARLY_WINDOW_HEIGHT("earlyWindowHeight", 480, "Early window height"),
+        EARLY_WINDOW_GL_VERSIONS("earlyWindowGLVersions", List.of("4.6", "4.1", "3.2"), "GL versions to try and bootstrap")
```

Comment:
This should include a full list from 4.6 down, as there are GPUs that support specifically 4.5 (AMD/Intel), 4.4 (Intel), 4.3 (Intel), 4.2 (Intel), 4.0 (Intel), or 3.3 (anything that supports 3.2). The drivers are free to return these versions anyway, but not all will (Intel won't on windows, Mesa will) so the latest possible should be asked for.

Reply:
It's intended as a config to try and address this.

Reply:
Unless its causing a bug/crash, this config value will never be touched, which is why I believe that it should start with the full list. The most widely used GPUs are also the most problematic with this truncated list as Intel on windows supports 4.6, 4.4, 4.3, or 4.0. with 4.3+ all making some extremely useful extensions core. For AMD its the TeraScale GPUs that only have 4.5 support. That could leave a ton of very useful features completely unused.

Regardless of the 4.x versions though, at a minimum 3.3 should be added as _all_ 3.2 capable GPUs do support 3.3, with the exception of ancient (> 10 years) drivers. With the most critical addition being ARB_instanced_arrays.

Reply:
3.2 is what Minecraft itself uses. Having a config value will make startup a bit more efficient by removing unnecessary checks, in my opinion, even if we config in the full set.

Reply:
I understand needing to have 3.2 as an option, I'm not arguing for that being removed. I want 3.3 added to the defaults as that check will never actually fail, but still leave 3.2 there because you cant fully rely on _trust me bro_ for that.

For efficiency, how much of a startup time difference will a few extra attempts at this actually make for startup time? I highly doubt its even close to enough to matter.

Reply:
I wanted config because of new versions. But apparently there's not new versions of gl. So that's pointless. Heh.

Reply:
Switched to a fixed list. It appears GL is endoflife heh.